### ISSUE TITLE
Misleading DB Error

### ISSUE THREAD

**Issue by craigem** [input-output-hk/cardano-byron-proxy#77]
If a node is in the static routes that is not in the topology file, a database error is reported instead of the missing node:

```
Started byron-proxy node service.
[cardano_byron_proxy.#messagecounters.ekgview:Debug:15] [2019-11-28 02:22:45.11 UTC] time_interval_(s) = 0
[cardano_byron_proxy.:Info:4] [2019-11-28 02:22:45.76 UTC] Opening database
[cardano_byron_proxy.index:Info:46] [2019-11-28 02:22:45.77 UTC] Rollback Origin
[cardano_byron_proxy.:Info:4] [2019-11-28 02:22:45.77 UTC] Database opened
cardano-byron-proxy: ExceptionInLinkedThread "ThreadId 46" ClosedDBError
byron-proxy.service: Main process exited, code=exited, status=1/FAILURE
```

**Comment by mrBliss** [comment]
Note that this `ClosedDBError` must be thrown when the ChainDB is being used *after it was closed because of an unrecoverable error*. My hypothesis is that the unrecoverable error is not being traced (either the tracer in question is not enabled or the trace message is being ignored). Moreover, the unrecoverable error should shut down the whole process, something which is not being doing properly, as the `ClosedDBError` indicates that there's still a thread running which tries to use the ChainDB.

As you can see, the process does terminate and I don't think much harm can be done this way, but it should at least trace the unrecoverable error that is being thrown.

**Comment by mrBliss** [comment]
See https://github.com/input-output-hk/ouroboros-network/issues/1350.

**Comment by mrBliss** [comment]
I believe this will be fixed when #89 is merged.

**Comment by oneEdoubleD** [comment]
I replicated this with @jbgi's help on the `dev-deployer` by adding a fake node to the static routes on the basis this would then not be in the `topology` file. The logs report a missing node rather than a DB error so I'd agree this has been fixed. 

Error:
`cardano-byron-proxy: UndefinedNodeName (NodeName "bogus-node")`

Full logs:
```
Feb 11 10:06:57 p-a-1 8f9p09bapjbwxpmp49zyp230kh2nw648-unit-script-byron-proxy-start[7415]: [cardano_byron_proxy.:Info:4] [2020-02-11 10:06:57.35 UTC] Database opened
Feb 11 10:06:57 p-a-1 8f9p09bapjbwxpmp49zyp230kh2nw648-unit-script-byron-proxy-start[7415]: [cardano_byron_proxy.index:Info:34] [2020-02-11 10:06:57.35 UTC] Rollback (At (Block {blockPointSlot = SlotNo {unSlotNo = 902213}, blockPointHash = ByronHash {unByronHash = AbstractHash 53af598ede2890129967e3ef1136aaadd883fdb5e80cc731a54f893c4b0ea461}}))
Feb 11 10:06:57 p-a-1 8f9p09bapjbwxpmp49zyp230kh2nw648-unit-script-byron-proxy-start[7415]: Shutting down..
Feb 11 10:06:57 p-a-1 8f9p09bapjbwxpmp49zyp230kh2nw648-unit-script-byron-proxy-start[7415]: cardano-byron-proxy: UndefinedNodeName (NodeName "bogus-node")
Feb 11 10:06:57 p-a-1 systemd[1]: byron-proxy.service: Main process exited, code=exited, status=1/FAILURE
Feb 11 10:06:57 p-a-1 systemd[1]: byron-proxy.service: Failed with result 'exit-code'.
```
